DM-2

mbelogubov opened the pull request:
Now list gets all addons.
Improve list appearance.
Add install and delete buttons.

belogubovm committed:
DM-2 : Now list gets all addons.
Improve list appearance.
Add install and delete buttons.

belogubovm committed:
Merge branch 'develop' of github.com:IvanProgrammer1703/developer_mode into DM-2

belogubovm committed:
DM-2 : Add favorite addons list.
Add search. Add settings.
Fix orders ajax adding product issue.

belogubovm committed:
DM-2 : Fix add to favorites.

belogubovm committed:
DM-2 : Add icons. Fix styles.

belogubovm committed:
DM-2 : Add icons. Fix styles.

IvanProgrammer1703 commented on app/addons/addon_developer/AddonHelper.php:
```diff
@@ -14,30 +14,16 @@
-        'banners',
-        'seo'
-    ];
+    // public static $favoriteAddons;
```
PhpDoc!!!

IvanProgrammer1703 commented on app/addons/addon_developer/controllers/backend/addons.post.php:
```diff
@@ -21,12 +22,74 @@
 
 if ($_SERVER['REQUEST_METHOD'] == 'POST') {
 
```
пробел не нужен

IvanProgrammer1703 commented on app/addons/addon_developer/controllers/backend/addons.post.php:
```diff
@@ -21,12 +22,74 @@
+        $params = $_REQUEST;
+        $addon_name = $params['addon_name'] ?? null;
+        $result = [];
+        if ($addon_name) {
```
после result

IvanProgrammer1703 replied:
перенос строки

IvanProgrammer1703 commented on app/addons/addon_developer/controllers/backend/addons.post.php:
```diff
@@ -21,12 +22,74 @@
+            }
+
+        }
+        return json_encode($result);
```
перед return всегда перенос строки

IvanProgrammer1703 commented on design/backend/templates/addons/addon_developer/hooks/index/actions.post.tpl:
```diff
@@ -1,13 +1,59 @@
+{* <div style="background: white; position: fixed;"> *}
+{* <div style="overflow-y: scroll; height: 200px;"> *}
+    <div style="padding: 2px 5px" class="form-inline object-selector object-addon-add cm-object-addon-add-container">
+        <select id="addon_select"
```
Перенсти ID

IvanProgrammer1703 commented on design/backend/templates/addons/addon_developer/hooks/index/actions.post.tpl:
```diff
@@ -1,13 +1,59 @@
+                { data-ca-autofocus="false" }
+            { else }
+                { data-ca-autofocus="true" }
+            { /if }>
```
">" ее тоже перенести

belogubovm committed:
Merge branch 'DM-2' of github.com:IvanProgrammer1703/developer_mode into DM-2

IvanProgrammer1703 commented on app/addons/addon_developer/AddonHelper.php:
```diff
@@ -14,30 +14,16 @@
 
     public static function getAddonList($params = [])
     {
-        $addons = Tygh::$app['db']->getArray('SELECT * FROM ?:addons WHERE addon IN (?a) LIMIT 0, ?i', static::$favoriteAddons, $params['count'] ?? 5);
```
Длинная строка, можно перенести.

belogubovm committed:
DM-2 : Fix controller issues.
Prettify appearance.
Add style classes.
Disable addon on addons manage page.

belogubovm committed:
DM-2 : Fix controller issues.
Prettify appearance.
Add style classes.
Disable addon on addons manage page.

mbelogubov committed:
DM-2 : Fix unclickable addons in search

belogubovm committed:
Merge branch 'DM-2' of github.com:IvanProgrammer1703/developer_mode into DM-2

belogubovm committed:
DM-2 : Add ajax adding addons to list.
Add enable/disable buttons.

belogubovm committed:
DM-2 : Search now contains only addons that are not present in favorites.
Fix duplicate addition in favorites.
Fix form in form bug.
Add cm-ajax to all links.
Auto-remove selected item from search.

belogubovm committed:
DM-2 : Add addon_dev controller.
Rename class AddonHelper to AddonDev.
All methods moved to AddonDev class.
Add addon state toggle.

belogubovm committed:
DM-2 : Fix adding to favorites.

belogubovm committed:
DM-2 : Add switch check if addon state changed.

belogubovm committed:
DM-2 : Add settings in popup.

belogubovm committed:
DM-2 : Add removing addon from favorites.
Fix bug when dropdown wasn't shown if empty favorites.
Fix switch style.
Add popup for self setings.
Move dropdown position right to fit in narrow viewport.

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+****************************************************************************
+* PLEASE READ THE FULL TEXT  OF THE SOFTWARE  LICENSE   AGREEMENT  IN  THE *
+* "copyright.txt" FILE PROVIDED WITH THIS DISTRIBUTION PACKAGE.            *
+****************************************************************************/
```
Надо определиться мы пишем эту штуку или нет.

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+use Tygh\Settings;
+use Tygh\Addons\SchemesManager;
+
+class AddonDev
```
phpDoc

IvanProgrammer1703 replied:
везде

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+
+    public static function installAddon($addon_id)
+    {
+        return fn_install_addon($addon_id);
```
Зачем ?

IvanProgrammer1703 replied:
Переопределяешь одну и туже функцию без именения функционала.

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+
+    public static function uninstallAddon($addon_id)
+    {
+        return fn_uninstall_addon($addon_id);
```
Тот же вопрос.

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+        $settings_values = fn_get_addon_settings_values($addon_id);
+        $settings_vendor_values = fn_get_addon_settings_vendor_values($addon_id);
+
+        $update_addon_settings_result = fn_update_addon_settings($addon_scheme, true, $settings_values, $settings_vendor_values);
```
Может перенести ? длинноватая строка(но тут не точно))

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+        Registry::clearCachedKeyValues();
+
+        if ($update_addon_settings_result) {
+            fn_set_notification('N', __('notice'), __('text_addon_refreshed', [
```
'N' => YesNo::No;

IvanProgrammer1703 replied:
Забыл переносы:
fn_set_notification(
YesNo::No,
__('notice'),
__('text_notice')
);

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+
+    public static function toggleAddon($addon_id, $state)
+    {
+        $state_changed = static::updateAddonStatus($addon_id, $state ? 'A' : 'D');
```
Можно не заводить переменную.

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+            $status = fn_update_addon_status($addon_id, $status);
+        }
+        Registry::clearCachedKeyValues();
+        return $status;
```
Нету строки перед return

IvanProgrammer1703 commented on app/addons/addon_developer/AddonDev.php:
```diff
@@ -0,0 +1,185 @@
+                'addon' => $addon_id,
+                'name' => $addon_info['name'],
+                'status' => $addon_info['status'],
+                'urls' => static::generateAddonUrls($addon_id, $addon_info['status'], $addon_info['has_options'], $return_url)
```
Вот тут можно тоже перенос сделать, строка длинноватая.(но не точно)

belogubovm committed:
DM-2 : Add updating list state after (un)installing addons.

belogubovm committed:
DM-2 : Encapsulate methods into AddonDev class.
Fix generateAddonUrl issues.
Improve init controller check.
Fix own settings url.
Improve addon_id transfering to template.
Fix show/hide no-items block.
Fix empty favorite addons list issues.
Refactor everything.
Fix default own setting.
Fix russian language.

belogubovm committed:
DM-2 : Fix menu and switch styles.

belogubovm committed:
DM-2 : Fix absent addon_id for tempalte.

belogubovm committed:
DM-2 : Remove icons hover background, add scale transform instead.
Clear code.

belogubovm committed:
DM-2 : Fix double request for remove from favorites.
Add checking return url to prevent recursive url encoding bug.
Add addon name parameter for remove from favorites notification.
Add switch margin and hover scale.

belogubovm committed:
DM-2 : Add phpdoc.
Fix styles.
Minor fixes.
